### ISSUE TITLE
ResizeObserver - loop limit exceeded

### ISSUE THREAD

**Issue by cornelius-behrend** [wellyshen/react-cool-dimensions#497]
# Bug Report

## Describe the Bug

Since Chrome 64, the ResizeObserver is firing a silent error that might be captured and raised when adding an event listener to the window (like some UI error handlers do).

See [this GitHub discussion](https://github.com/souporserious/react-measure/issues/104).

In index.ts, this can be fixed as follows:

```
// eslint-disable-next-line compat/compat
    observerRef.current = new (polyfill || window.ResizeObserver)(
      ([entry]: any) => {
        window.requestAnimationFrame(() => { /* < New line */
```

## How to Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Use Chrome 64 / Safari 14 or newer
2. `window.addEventListener('error', function(e) { console.log(e); });` to log the error

## Screenshot

<img width="485" alt="Bildschirmfoto 2021-06-07 um 17 17 22" src="https://user-images.githubusercontent.com/77003328/121042753-46280280-c7b4-11eb-8ed6-6e8f092083da.png">

## Expected Behavior

No errors from ResizeObserver should be forwarded to the event listeners.

## Your Environment

- Device: [MacBook Pro]
- OS: [macOS 11.4]
- Browser: [Chrome, Safari]
- Version: [91.0.4472.77, 14.1.1]

**Comment by wellyshen** [comment]
@cornelius-behrend What version that you're using? We have handled it (#373)

**Comment by cornelius-behrend** [comment]
@wellyshen thanks for the reply! I used `yarn add react-cool-dimensions`. So I'm on the most recent version (2.0.6). I fixed it locally with the code above.

**Comment by wellyshen** [comment]
@cornelius-behrend Seems you found the solution, could you mind to send me a PR for that?

**Comment by cornelius-behrend** [comment]
Sure! :)

**Comment by wellyshen** [comment]
@cornelius-behrend Thank you mate, it really helpful 💯

**Comment by cornelius-behrend** [comment]
@wellyshen I'm glad I could help. Hopefully, the problem will be fixed with that in every case. The PR is [here](https://github.com/wellyshen/react-cool-dimensions/pull/501).

**Comment by wellyshen** [comment]
@cornelius-behrend I'm wondering if we need to keep [the old solution](https://github.com/wellyshen/react-cool-dimensions/pull/501/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R169) or not , would you mind to confirm it in your reproduced environment for us?

**Comment by wellyshen** [comment]
@cornelius-behrend Thanks mate, it has been released via [v2.0.7](https://github.com/wellyshen/react-cool-dimensions/releases/tag/v2.0.7) 👍🏻
